stream.go

```diff
@@ -0,0 +1,439 @@
+	AccessRules  StreamAccessRules `json:"accessRules,omitempty"`
+}
+
+// StreamVideoResponse represents an API response of a stream video.
+type StreamVideoResponse struct {
+	Response
+	Result StreamVideo `json:"result,omitempty"`
+}
+
+// StreamVideoCreateResponse represents an API response of creating a stream video.
+type StreamVideoCreateResponse struct {
+	Response
+	Result StreamVideoCreate `json:"result,omitempty"`
+}
+
+// StreamListResponse represents the API response from a StreamListRequest.
+type StreamListResponse struct {
+	Response
+	Result []StreamVideo `json:"result,omitempty"`
+	Total  string        `json:"total,omitempty"`
+	Range  string        `json:"range,omitempty"`
+}
+
+// StreamSignedURLResponse represents an API response for a signed URL.
+type StreamSignedURLResponse struct {
+	Response
+	Result struct {
+		Token string `json:"token,omitempty"`
+	}
+}
+
+// StreamAccessRules represents the accessRules when creating a signed URL.
+type StreamAccessRules []map[string]interface{}
+
+// StreamUploadFromURL send a video URL to it will be downloaded and made available on Stream.
+//
+// API Reference: https://api.cloudflare.com/#stream-videos-upload-a-video-from-a-url
+func (api *API) StreamUploadFromURL(ctx context.Context, params StreamUploadFromURLParameters) (StreamVideo, error) {
+	if params.AccountID == "" {
+		return StreamVideo{}, ErrMissingAccountID
+	}
+
+	if params.URL == "" {
+		return StreamVideo{}, ErrMissingUploadURL
+	}
+
+	uri := fmt.Sprintf("/accounts/%s/stream/copy", params.AccountID)
+
+	res, err := api.makeRequestContext(ctx, http.MethodPost, uri, params)
+	if err != nil {
+		return StreamVideo{}, err
+	}
+
+	var streamVideoResponse StreamVideoResponse
+	if err := json.Unmarshal(res, &streamVideoResponse); err != nil {
+		return StreamVideo{}, err
+	}
+	return streamVideoResponse.Result, nil
+}
+
+// StreamUploadVideoFile uploads a video from a path to the file.
+//
+// API Reference: https://api.cloudflare.com/#stream-videos-upload-a-video-using-a-single-http-request
+func (api *API) StreamUploadVideoFile(ctx context.Context, params StreamUploadFileParameters) (StreamVideo, error) {
+	if params.AccountID == "" {
+		return StreamVideo{}, ErrMissingAccountID
+	}
+
+	if params.FilePath == "" {
+		return StreamVideo{}, ErrMissingFilePath
+	}
+
+	uri := fmt.Sprintf("/accounts/%s/stream", params.AccountID)
+
+	// Create new multipart writer
+	body := &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+	formFile, err := writer.CreateFormFile("file", params.FilePath)
+	if err != nil {
+		return StreamVideo{}, err
+	}
+	file, err := os.Open(params.FilePath)
+	if err != nil {
+		return StreamVideo{}, err
+	}
+	if _, err := io.Copy(formFile, file); err != nil {
+		return StreamVideo{}, err
+	}
+	if err := writer.Close(); err != nil {
+		return StreamVideo{}, err
+	}
+
+	res, err := api.makeRequestWithAuthTypeAndHeaders(ctx, http.MethodPost, uri, body, api.authType, http.Header{
```

Comment:
we'll want to use `makeRequestContextWithHeaders` instead of `makeRequestWithAuthTypeAndHeaders` (which is the underlying method, not abstracted)
 
```suggestion
	res, err := api.makeRequestContextWithHeaders(ctx, http.MethodPost, uri, body, http.Header{
```